Add L1 and L2 category support

GideonBear opened the pull request:
Whoops, I forgot I needed this as well. Here's another pull request :)
All L1 and L2 categories are stored in two JSON files, which can be created using the `scrape_categories.py` script. They might need periodic updating, which can be done with the `update_categories.py` script.
Categories are created using their name, to avoid invalid categories/id's from being used. It will give a `ValueError` if the category does not exist.

GideonBear commented:
I've also added a utility function `category_from_name` to get an L1 or L2 category.

jensjeflensje commented:
Hi, thanks for taking the time to make a script to fetch the categories! This is a nice addition.
The main thing I'm concerned with is how much these categories (and their names) are going to change. We might want to include a `from_id` method too to make sure a program would still work after marktplaats changes their labels.
I just wanted to hear your thoughts on this one.

GideonBear commented:
> Hi, thanks for taking the time to make a script to fetch the categories! This is a nice addition. The main thing I'm concerned with is how much these categories (and their names) are going to change. We might want to include a `from_id` function too to make sure a program would still work after marktplaats changes their labels. I just wanted to hear your thoughts on this one.

This was originally my plan (hence the `from_name` naming) but I decided not to add it because of the automatic L1 category addition via the `parent`.
At first I was going to add two parameters to `SearchQuery`, but this is obviously much simpler.

Do you think that `L1Category.from_id(id: int)` and `L2Category.from_id(id: int, parent: L1Category)` is a good idea? That seems like the most elegant solution here.

GideonBear commented:
That way you can:
* Create an `L2Category` from its id and its parents id
* Create an `L2Category` from its id and its parents name
but still not do an API call with just the `l2Category`.

jensjeflensje commented:
I feel like the 2 from_id functions would be nice for more novel users. I'm hoping people will never have to construct a category class themselves.

It would be nicest to be able to get the L1 category from the L2, but that might be outside scope for now :)

GideonBear commented:
> I feel like the 2 from_id functions would be nice for more novel users. I'm hoping people will never have to construct a category class themselves.
> 
> It would be nicest to be able to get the L1 category from the L2, but that might be outside scope for now :)

Added! I added the option to specify a name as well, but left it optional, opting for `"Unkown"` as the default.

jensjeflensje commented:
Great! I'll merge it into the `pre-release` branch right now, and merge it with main once I've processed the other PRs :)
Thanks again!
